File: sampleCode/CSharp/ConsoleApp/Endpoints/Events/SpendingPatternCommodity.cs
namespace ConsoleApp.Endpoints.Events;

public sealed record class SpendingPatternCommodity
{
    public double? Coefficient { get; set; }
    public int CommodityCode { get; set; }
    public string CommodityDescription { get; set; }
    public bool IsSamValue { get; set; }
    public bool IsUserCoefficient { get; set; }
    public double LocalPurchasePercentage { get; set; } = 1.0d; // 100%
}

File: sampleCode/Python/models/spending_pattern_commodity.py
class SpendingPatternCommodity:
    """
    The Model for a Spending Pattern Commodity
    """

    def __init__(self,
                 coefficient: float | None,
                 commodity_code: int,
                 commodity_description: str,
                 is_sam_value: bool,
                 is_user_coefficient: bool,
                 local_purchase_percentage: float = 1.0):
        self.coefficient = coefficient
        self.commodity_code = commodity_code
        self.commodity_description = commodity_description
        self.is_sam_value = is_sam_value
        self.is_user_coefficient = is_user_coefficient
        self.local_purchase_percentage = local_purchase_percentage
